src/app/zap-templates/partials/cluster-objects-struct.zapt

```diff
@@ -113,32 +113,28 @@ CHIP_ERROR Type::Encode(TLV::TLVWriter & aWriter, TLV::Tag aTag) const
 {{/if}}
 
 CHIP_ERROR DecodableType::Decode(TLV::TLVReader &reader) {
-    CHIP_ERROR err = CHIP_NO_ERROR;
-    TLV::TLVType outer;
-    VerifyOrReturnError(TLV::kTLVType_Structure == reader.GetType(), CHIP_ERROR_WRONG_TLV_TYPE);
-    err = reader.EnterContainer(outer);
-    ReturnErrorOnFailure(err);
-    while ((err = reader.Next()) == CHIP_NO_ERROR) {
-        if (!TLV::IsContextTag(reader.GetTag()))
+    detail::StructDecodeIterator __iterator(reader);
+    while (true) {
+        detail::StructDecodeIterator::EntryElement __element = __iterator.Next();
+        if (std::holds_alternative<CHIP_ERROR>(__element)) {
+           return std::get<CHIP_ERROR>(__element);
+        }
+
+        CHIP_ERROR err = CHIP_NO_ERROR;
+        [[maybe_unused]] const uint8_t __context_tag = std::get<uint8_t>(__element);
```

Comment:
Using this actually saves a few more 100s of bytes for empty commands. Applied.

Reply:
_That_ would be a thing worth documenting.  Say what we measured and on what platform, and then people will know they should re-measure before changing.....

I am pretty surprised this is smaller, by the way!

Reply:
Added a comment. My tests were on `efr32-brd4161a-light-no-version`.

I just re-tested, changing commands:
  - with switch: 972,328 size for `.text`
  - with if/else: 971,912 size for `.text`
  
  I also found it quite counter-intuitive. I liked switch because it results in a smaller CPP file due to fewer `{}` ...